src/textual/app.py

```diff
@@ -3275,6 +3275,15 @@ async def _on_app_blur(self, event: events.AppBlur) -> None:
         self.app_focus = False
         self.screen.refresh_bindings()
 
+    def _is_dark_color(self, r: int, g: int, b: int) -> bool:
+        # perceived brightness formula
+        perceived_brightness = 0.2126 * r + 0.7152 * g + 0.0722 * b
+        return perceived_brightness < 128
+
+    async def _on_background_color(self, event: events.BackgroundColor) -> None:
```

Comment:
I wouldn't want to do this automatically, unless the dev explicitly requests this behaviour.

Sending the message is probably fine, until we have a mechanism for the dev to request automatic light / dark mode.